tests/send_and_receive.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright 2016 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under (1) the MaidSafe.net Commercial License,
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the SAFE Network Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.0.  This, along with the
+// Licenses can be found in the root directory of this project at LICENSE, COPYING and CONTRIBUTOR.
+//
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+//
+// Please review the Licences for the specific language governing permissions and limitations
+// relating to use of the SAFE Network Software.
+
+extern crate crust;
+#[macro_use]
+extern crate maidsafe_utilities;
+
+use crust::{CrustEventSender, Event, Service};
+use maidsafe_utilities::event_sender::{MaidSafeEventCategory, MaidSafeObserver};
+use std::sync::mpsc::{self, Receiver};
+use std::thread::{self, JoinHandle};
+
+const BEACON_PORT: u16 = 44444;
```

Comment:
Using hard-coded values for ports in tests has previously caused problems when the port has been unavailable on the test machine (often because another test has acquired the port).  Is it OK here?